osgtest/library/files.py

```diff
@@ -144,12 +144,9 @@ def replace(path, old_line, new_line, owner=None, backup=True):
     write(path, lines_to_write, owner, backup)
 
 def replace_regexpr(path, regexp, new_line, owner=None, backup=True):
-    """Replace an old line that matches regexpr with a new line in given path.                                                     
-    The 'owner' and 'backup' arguments are passed to write().                                                                                  
+    """Replace an old line that matches regexpr with a new line in given path.
+    The 'owner' and 'backup' arguments are passed to write().
     """
-    match = re.search(regexp, new_line)
-    if not match:
-       raise ValueError("The new_line '%s' does not match reg expr '%s'" % (new_line, regexp)) 
     lines_to_write = []
```

Comment:
I don't see any real use of the input verification here and I think it only causes developer headache: "I want to replace this regular expression in this file with this new line. Wait, the new line has to match the regular expression? What? Why? I want to replace it with whatever I damn well please." That was my response, at least. I've already gone through the rest of the test code to look at usage of this function but since my change lifts the requirement, it doesn't have any affect.

That being said, I could revert it and add the option but I think it's an unnecessary and confusing 'feature' of the function.